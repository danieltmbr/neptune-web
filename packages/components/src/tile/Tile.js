import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';

import './Tile.css';

export const Tile = ({ className, title, content, media, onClick }) => {
  return (
    <div className={classNames('tw-tile', className)}>
      <div className="m-y-2 m-x-auto">{media}</div>
      {/* TODO: It complains about using anchor as a button... */}
      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
      <a
        className="tw-tile__link"
        onClick={onClick}
        tabIndex="0"
        role="button"
        onKeyDown={(e) => {
          if (e.key === 'Enter' || e.key === 'Spacebar') {
            onClick();
          }
        }}
      >
        <h4 className="m-b-1">{title}</h4>
      </a>
      {content}
    </div>
  );
};

Tile.propTypes = {
  title: Types.node.isRequired,
  content: Types.node,
  media: Types.node,
  onClick: Types.func.isRequired,
  className: Types.string,
};

Tile.defaultProps = {
  media: null,
  content: null,
  className: null,
};

export default Tile;
