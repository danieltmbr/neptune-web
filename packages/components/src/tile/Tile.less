.tw-tile {
  position: relative;
  border: 1px solid var(--color-border-neutral);
  border-radius: 3px;

  &--large {
    width: 340px;
  }

  &--small {
    width: 180px;

    .h4 {
      font-size: var(--font-size);
    }

    > .tw-tile__content {
      font-size: var(--font-size-small);
    }
  }

  &__link {
    &:focus {
      outline: none;
    }

    &::after {
      content: '';
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
    }
  }

  &:hover,
  &:focus-within {
    border: 1px solid var(--color-border-focus);
    box-shadow: 0px 0px 6px rgba(0, 0, 0, 0.2);
  }

  & + .tw-tile {
    margin-left: 24px;
  }
}
