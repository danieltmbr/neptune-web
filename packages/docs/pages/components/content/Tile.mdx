import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import tileCode from '../../../liveEditorCode/tile.code';
import pickerCode from '../../../liveEditorCode/picker.code';

import { Tile, Picker } from '@transferwise/components';

<LiveEditorBlock code={tileCode} scope={{ Tile }} />
<GeneratePropsTable componentName="Tile" />

## Usage
TODO: do we need to add information that this is used for major decisions where more context is needed?
E.g. deciding between two options

## Groups of tiles

TODO: Add information about the picker

TODO: Add notice for not showing more than 2-3 tiles for decision

<LiveEditorBlock code={pickerCode} scope={{ Picker }} display="vertical" />
<GeneratePropsTable componentName="Picker" />


export const meta = {
  name: 'Tile',
}