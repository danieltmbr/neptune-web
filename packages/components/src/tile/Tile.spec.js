import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Tile from '.';

describe(Tile, () => {
  const onClick = jest.fn();
  const defaultProps = {
    title: 'Receive money',
    content: <span>Receive money from friends and familty like a local.</span>,
    media: <svg data-testid="illustration" />,
    onClick: () => onClick(),
  };

  beforeEach(() => jest.clearAllMocks());

  it('matches snapshot when regular size', () => {
    const { asFragment } = render(<Tile {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when small size', () => {
    const { asFragment } = render(<Tile size={Tile.Size.SMALL} {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick when anchor is clicked', async () => {
    render(<Tile {...defaultProps} />);

    expect(onClick).not.toHaveBeenCalled();

    const title = await screen.findByText('Receive money');

    expect(onClick).not.toHaveBeenCalled();

    fireEvent.click(title);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
