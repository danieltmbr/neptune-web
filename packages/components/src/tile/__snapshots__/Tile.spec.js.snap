// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile matches snapshot when regular size 1`] = `
<DocumentFragment>
  <div
    class="tw-tile d-flex flex-column text-xs-center tw-tile--large p-x-5 p-y-4"
  >
    <div
      class="m-y-2 m-x-auto"
    >
      <svg
        data-testid="illustration"
      />
    </div>
    <a
      class="tw-tile__link text-no-decoration "
      role="button"
      tabindex="0"
    >
      <span
        class="h4"
      >
        Receive money
      </span>
    </a>
    <div
      class="tw-tile__content"
    >
      <span>
        Receive money from friends and familty like a local.
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tile matches snapshot when small size 1`] = `
<DocumentFragment>
  <div
    class="tw-tile d-flex flex-column text-xs-center tw-tile--small p-a-2"
  >
    <div
      class="m-y-2 m-x-auto"
    >
      <svg
        data-testid="illustration"
      />
    </div>
    <a
      class="tw-tile__link text-no-decoration "
      role="button"
      tabindex="0"
    >
      <span
        class="h5"
      >
        Receive money
      </span>
    </a>
    <div
      class="tw-tile__content"
    >
      <span>
        Receive money from friends and familty like a local.
      </span>
    </div>
  </div>
</DocumentFragment>
`;
